File: treatment/src/test/java/com/hartwig/actin/treatment/ctc/InterpretedCohortStatusComparatorTest.java
package com.hartwig.actin.treatment.ctc;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

public class InterpretedCohortStatusComparatorTest {

    @Test
    public void shouldRankCohortStatesCorrectly() {
        InterpretedCohortStatus openWithSlots = ImmutableInterpretedCohortStatus.builder().open(true).slotsAvailable(true).build();
        InterpretedCohortStatus openWithoutSlots = ImmutableInterpretedCohortStatus.builder().open(true).slotsAvailable(false).build();
        InterpretedCohortStatus closedWithoutSlots = ImmutableInterpretedCohortStatus.builder().open(false).slotsAvailable(false).build();

        List<InterpretedCohortStatus> states = new ArrayList<>();
        states.add(openWithoutSlots);
        states.add(openWithSlots);
        states.add(closedWithoutSlots);

        states.sort(new InterpretedCohortStatusComparator());

        assertEquals(closedWithoutSlots, states.get(0));
        assertEquals(openWithoutSlots, states.get(1));
        assertEquals(openWithSlots, states.get(2));
    }

}

File: trial/src/test/kotlin/com/hartwig/actin/trial/status/InterpretedCohortStatusComparatorTest.kt
package com.hartwig.actin.trial.status

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class InterpretedCohortStatusComparatorTest {

    @Test
    fun shouldRankCohortStatesCorrectly() {
        val openWithSlots = InterpretedCohortStatus(open = true, slotsAvailable = true)
        val openWithoutSlots = InterpretedCohortStatus(open = true, slotsAvailable = false)
        val closedWithoutSlots = InterpretedCohortStatus(open = false, slotsAvailable = false)

        val states = listOf(openWithoutSlots, openWithSlots, closedWithoutSlots).sortedWith(InterpretedCohortStatusComparator())
        assertThat(states).containsExactly(closedWithoutSlots, openWithoutSlots, openWithSlots)
    }
}